pkg/repro/repro.go

```diff
@@ -421,13 +435,18 @@ func (ctx *context) simplifyProg(res *Result) (*Result, error) {
 		ctx.stats.SimplifyProgTime = time.Since(start)
 	}()
 
-	opts := res.Opts
-	opts.Collide = false
-	crashed, err := ctx.testProg(res.Prog, res.Duration, opts)
-	if err != nil {
-		return res, err
+	if res.ops.Collide {
+		opts := res.Opts
+		opts.Collide = false
+		crashed, err := ctx.testProg(res.Prog, res.Duration, opts)
+		if err != nil {
+			return res, err
+		}
+		if crashed {
+			res.Opts = opts
+		}
 	}
-	if crashed {
+	if !res.ops.Collide && res.ops.Threaded {
 		res.Opts = opts
 		opts.Threaded = false
 		crashed, err := ctx.testProg(res.Prog, res.Duration, opts)
```

Comment:
It looks like the right time to unify code in simplifyC and simplifyProg. There is too much duplication between these functions and in these functions.